sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -145,49 +146,71 @@ public void showLabMessagesSlider(List<LabMessageDto> labMessages) {
-					caseSimilarityCriteria.personUuid(selectedPerson.getUuid());
-					List<CaseSelectionDto> similarCases = FacadeProvider.getCaseFacade().getSimilarCases(caseSimilarityCriteria);
-
-					ContactSimilarityCriteria contactSimilarityCriteria = new ContactSimilarityCriteria();
-					contactSimilarityCriteria.setPerson(selectedPerson);
-					contactSimilarityCriteria.setDisease(labMessageDto.getTestedDisease());
-					List<SimilarContactDto> similarContacts = FacadeProvider.getContactFacade().getMatchingContacts(contactSimilarityCriteria);
+	}
 
-					EventParticipantCriteria eventParticipantCriteria = new EventParticipantCriteria();
-					eventParticipantCriteria.setPerson(selectedPerson);
-					eventParticipantCriteria.setDisease(labMessageDto.getTestedDisease());
-					List<SimilarEventParticipantDto> similarEventParticipants =
-						FacadeProvider.getEventParticipantFacade().getMatchingEventParticipants(eventParticipantCriteria);
+	private void checkRelatedForwardedMessages(LabMessageDto labMessage) {
+		if (FacadeProvider.getLabMessageFacade().existsForwardedLabMessageWith(labMessage.getReportId())) {
+			VaadinUiUtil.showConfirmationPopup(
+				I18nProperties.getCaption(Captions.labMessageForwardedMessageFound),
+				new Label(I18nProperties.getString(Strings.messageForwardedLabMessageFound)),
+				I18nProperties.getCaption(Captions.actionYes),
+				I18nProperties.getCaption(Captions.actionCancel),
+				null,
+				yes -> {
+					if (yes) {
+						checkShortcuts(labMessage);
+					}
+				});
+		} else {
+			checkShortcuts(labMessage);
+		}
+	}
 
-					pickOrCreateEntry(labMessageDto, similarCases, similarContacts, similarEventParticipants, selectedPersonDto);
+	private void checkShortcuts(LabMessageDto labMessage) {
+		String labSampleId = labMessage.getLabSampleId();
+		if (labSampleId != null && !"".equals(labSampleId)) {
+			List<SampleDto> relatedSamples = FacadeProvider.getSampleFacade().getByLabSampleId(labSampleId);
+			SampleDto relatedSample = relatedSamples.size() == 1 ? relatedSamples.get(0) : null;
+			// if shortcut possible
+			if (relatedSample != null) {
+				List<LabMessageDto> relatedLabMessages = null;
+				// determine related messages
+				String reportId = labMessage.getReportId();
+				if (reportId != null && !"".equals(reportId)) {
+					relatedLabMessages = FacadeProvider.getLabMessageFacade()
+						.getForSample(relatedSample.toReference())
+						.stream()
+						.filter(
+							otherLabMessage -> reportId.equals(otherLabMessage.getReportId())
+								&& LabMessageStatus.PROCESSED.equals(otherLabMessage.getStatus()))
+						.collect(Collectors.toList());
 				}
-			}, false);
+
+				// propose to take shortcut
+				String message = (relatedLabMessages != null && !relatedLabMessages.isEmpty())
+					? I18nProperties.getString(Strings.messageRelatedSampleAndLabMessagesFound)
+					: I18nProperties.getString(Strings.messageRelatedSampleFound);
+				VaadinUiUtil.showChooseOptionPopup(
```

Comment:
There are the same options for both with and without related lab messages, does it make sense to get the related lab messages from the backend just for showing a slightly different message?

Reply:
I wondered about that, too. My conclusion was that it is necessary to show different messages here. Users might not want to directly jump to that sample when no related processed lab message could be found. I the sample edit  form, they don't have a way of seeing the related person data.

I also thought about creating some kind of extended LabMessageReferenceDto containing the reportId and the labMessageStatus, but discarded this because it requires quite some code for a small gain.

Are you okay with leaving it like that?

Reply:
It will probably be needed for correction too so it can be kept